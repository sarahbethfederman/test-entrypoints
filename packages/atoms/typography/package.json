{
  "name": "@auscred/typography",
  "version": "0.0.1",
  "main": "dist/Typography.js",
  "types": "dist/Typography.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
