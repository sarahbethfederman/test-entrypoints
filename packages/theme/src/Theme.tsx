import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
