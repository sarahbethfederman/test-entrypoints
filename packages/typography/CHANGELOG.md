## [4.1.0] 2020-06-24 [Release Notes]
- Added 'xl' as a size for Body

## [4.0.1] 2019-09-26 [Release Notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)
- Bumped lendi-ui dependencies to latest versions

## [4.0.0] 2019-9-23 2019-8-21 [Release notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)
### Changed
- Uses Theme v7
- Update typescript to version 3.5.2 and react to version 16.8.4 [HUB-151](https://creditandfinance.atlassian.net/browse/HUB-151)

## [3.4.0] - 2019-07-29 - [HUB-338](https://creditandfinance.atlassian.net/browse/HUB-338)
### Added
- Add 'xxs' type of Body, Heading, Link and Overline.

## [3.3.0] - 2019-07-16 - [HUB-330](https://creditandfinance.atlassian.net/browse/HUB-330)
### Changed
- Change link default color to 'currentColor'

## [3.2.1] - 2019-06-4 - [HUB-262](https://creditandfinance.atlassian.net/browse/HUB-262)
### Changed
- Updated utils dependency to get correct types

## [3.2.0] - 2019-05-27 - [HUB-169](https://creditandfinance.atlassian.net/browse/HUB-169)
### Changed
- Support for native props like Aria attributes, Standard HTML Attributes like title, classname, id, role, itemProp, itemID, itemRef

## [3.1.1] - 2019-05-23 - [HUB-201](https://creditandfinance.atlassian.net/browse/HUB-201)
### Changed
- Updated links that are set as a button to inherit background-color from parent.

## [3.1.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)
### Changed
- Module format reverted back to CJS.
### Deleted
- Dropped "module" property from package.json.

## [3.0.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)
### Changed
- Updated CHANGELOG to match required parsing format.

## [3.0.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)
### Changed
- Change module main entry point to esm in package.json

## [3.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)
### Deleted
- Remove the CJS bundles.

## [2.7.1] - 2019-04-10 - [HUB-50](https://creditandfinance.atlassian.net/browse/HUB-50)
### Changed
- Fixed the margin bug for Overline and added some tests for margin alignment in all typography.

## [2.7.0] - 2019-04-10 - [HUB-90](https://creditandfinance.atlassian.net/browse/HUB-90)
### Added
- Added "xl" size variant.

## [2.6.0] - 2019-03-14 - [BRAVO-571](https://creditandfinance.atlassian.net/browse/BRAVO-571)
### Added
- Added `Overline` sub-package, which includes `overline` mixin and `Overline` component.

## [2.5.5] - 2019-03-07 - [BRAVO-652](https://creditandfinance.atlassian.net/browse/BRAVO-652)
### Changed
- fixed bug that caused the error: 'children' doesn't exist on type 'IntrinsicAttributes & LinkProps'
