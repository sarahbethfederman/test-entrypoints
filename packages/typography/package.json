{
  "name": "@lendi-ui/typography",
  "version": "6.0.0-beta.0",
  "source": "src/index.ts",
  "main": "dist/typography.cjs.js",
  "module": "dist/typography.esm.js",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^10.0.0-beta.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^6.0.0-beta.0",
    "@lendi-ui/color": "^6.0.0-beta.0",
    "@lendi-ui/spacing": "^7.1.0-beta.0",
    "@lendi-ui/utils": "^6.0.0-beta.0"
  },
  "devDependencies": {
    "@lendi-ui/theme": "10.0.0-beta.0",
    "@types/react": "^16.4.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.9.0",
    "styled-components": "^4"
  }
}
