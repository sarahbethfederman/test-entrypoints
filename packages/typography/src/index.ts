import styled, { css } from 'styled-components';

import { select } from '@lendi-ui/theme';
import { gte, map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';
import { fg } from '@lendi-ui/color';

export type Alignment = 'left' | 'center' | 'right' | 'justify';
export type AlignmentOrAlignmentMap = BreakpointValue<Alignment> | BreakpointValueMap<Alignment>;
export type HeadingSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
export type BodySize = 'xs' | 'sm' | 'md' | 'lg';
export type LinkSize = 'sm' | 'md' | 'lg';

const alignMixin = (alignment: AlignmentOrAlignmentMap) => {
  return map(alignment, (val) => {
    if (val) {
      return `text-align: ${val};`;
    } else {
      return '';
    }
  });
};

const headingSizeMixin = (size: HeadingSize) => {
  switch (size) {
    case 'xs':
      return css`
        font-size: 15px;
        font-weight: 500;
        line-height: calc(20 / 15);

        ${gte('tablet')`
          font-size: 18px;
          line-height: calc(24/18);
        `};
      `;
    case 'sm':
      return css`
        font-size: 18px;
        font-weight: 500;
        line-height: calc(24 / 18);

        ${gte('tablet')`
          font-size: 22px;
          line-height: calc(28/22);
        `};
      `;
    case 'md':
      return css`
        font-size: 22px;
        font-weight: 500;
        line-height: calc(28 / 22);

        ${gte('tablet')`
          font-size: 28px;
          line-height: calc(32/28);
        `};
      `;
    case 'lg':
      return css`
        font-size: 28px;
        font-weight: 500;
        line-height: calc(34 / 28);

        ${gte('tablet')`
          font-size: 37px;
          line-height: calc(40/37);
        `};
      `;
    case 'xl':
      return css`
        font-size: 37px;
        font-weight: bold;
        line-height: calc(40 / 37);

        ${gte('tablet')`
          font-size: 46px;
          line-height: calc(50/46);
        `};
      `;
  }
};

const bodySizeMixin = (size: BodySize) => {
  switch (size) {
    case 'xs':
      return css`
        font-size: 10px;
        line-height: calc(16 / 10);

        ${gte('tablet')`
          font-size: 13px;
          line-height: calc(20/13);
        `};
      `;
    case 'sm':
      return css`
        font-size: 12px;
        line-height: calc(16 / 12);

        ${gte('tablet')`
          font-size: 14px;
          line-height: calc(20/14);
        `};
      `;
    case 'md':
      return css`
        font-size: 14px;
        line-height: calc(20 / 14);

        ${gte('tablet')`
          font-size: 16px;
          line-height: calc(24/16);
        `};
      `;
    case 'lg':
      return css`
        font-size: 16px;
        line-height: calc(24 / 16);

        ${gte('tablet')`
          font-size: 18px;
          line-height: calc(24/18);
        `};
      `;
  }
};

export interface HeadingOptions {
  color?: string;
  align?: Alignment;
  size: HeadingSize;
}

export function heading(options: HeadingOptions) {
  const { color = 'shade.700', align, size } = options;
  return css`
    ${color && fg(color)}
    ${align && alignMixin(align)}
    ${headingSizeMixin(size)}
    font-family: ${select('typography.heading.fontFamily', '"Cabin", sans-serif')};
  `;
}

export interface BodyOptions {
  color?: string;
  align?: Alignment;
  size?: BodySize;
}

/**
 * Body mixin
 */
export function body(options: BodyOptions = {}) {
  const { color = 'shade.700', align, size = 'md' } = options;
  return css`
    ${color && fg(color)}
    ${align && alignMixin(align)}
    ${size && bodySizeMixin(size)}
    font-family: ${select('typography.body.fontFamily', '"Open sans", sans-serif')};
  `;
}

/**
 * Body component
 */
export const Body = styled.p`
  ${link};
`;

export interface LinkOptions {
  color?: string;
  href?: string;
  size?: LinkSize;
}

/**
 * Link mixin
 */
export function link(options: LinkOptions = {}) {
  const { color = 'primary.500', size } = options;

  return css`
    ${color && fg(color)}
    ${size && bodySizeMixin(size)}
    font-family: ${select('typography.body.fontFamily', '"Open sans", sans-serif')};
    font-weight: bold;
    cursor: pointer;

    &:hover,
    &:focus,
    &:active {
      text-decoration: underline;
      ${fg('primary.700')}
    }
  `;
}

/**
 * Link component
 */
export const Link = styled.a`
  ${link};
`;
