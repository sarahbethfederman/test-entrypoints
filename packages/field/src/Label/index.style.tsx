import styled from 'styled-components';
import { Body, Heading } from '@lendi-ui/typography';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';
import { deriveSize } from '@lendi-ui/utils';

export type Size = 'lg' | 'sm';
export type LabelSize = BreakpointValue<Size> | BreakpointValueMap<Size>;

const bodyStyleBySizeMixin = (size: LabelSize) =>
  map(size, (val) => {
    switch (val) {
      case 'sm':
        return `
          font-size: ${deriveSize(0.75)};
      `;
      case 'lg':
        return `
          font-size: ${deriveSize(1)};
      `;
      default:
        return undefined;
    }
  });

const headingStyleBySizeMixin = (size: LabelSize) =>
  map(size, (val) => {
    switch (val) {
      case 'sm':
        return `
          font-size: ${deriveSize(1)};
          margin-right: ${deriveSize(0.5)};
      `;
      case 'lg':
        return `
          font-size: ${deriveSize(1.75)};
          margin-right: ${deriveSize(1)};
      `;
      default:
        return undefined;
    }
  });

export const LabelHeading = styled(Heading)`
  ${({ size }: { size: LabelSize }) => headingStyleBySizeMixin(size)} display: inline-block;
`;

export const AssistiveBody = styled(Body)`
  ${({ size }: { size: LabelSize }) => bodyStyleBySizeMixin(size)};
`;
