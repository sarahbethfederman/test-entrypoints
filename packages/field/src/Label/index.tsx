import * as React from 'react';
import { LinkProps } from '@lendi-ui/typography';
import { LabelSize, LabelBody, LabelHeading, AssistiveBody } from './index.style';

export interface LabelProps {
  size?: LabelSize;
  label?: string;
  assistiveText?: string;
  link?: React.ReactElement<LinkProps>;
  isOptional?: boolean;
}

const Label = ({ size = 'sm', label, assistiveText, link, isOptional }: LabelProps) => {
  let labelBody, assistiveBody;
  switch (size) {
    case 'sm':
      labelBody = (
        <LabelBody size="md" ml="xxs">
          <strong>
            {label}
            {isOptional && ' (Optional)'}
          </strong>
        </LabelBody>
      );
      assistiveBody = assistiveText ? (
        <AssistiveBody color="shade.400" size="xs">
          {assistiveText}
        </AssistiveBody>
      ) : null;
      break;
    case 'lg':
      labelBody = (
        <LabelHeading size="md">
          {label}
          {isOptional && ' (Optional)'}
        </LabelHeading>
      );
      assistiveBody = assistiveText ? (
        <AssistiveBody color="shade.400" size="md">
          {assistiveText}
        </AssistiveBody>
      ) : null;
      break;
  }
  return (
    <div>
      {labelBody}
      {link}
      {assistiveBody}
    </div>
  );
};

export default Label;
