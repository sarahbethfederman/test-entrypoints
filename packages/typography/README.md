# @lendi-ui/typography

Typography including Body, Heading and Link.

## Installation

```
yarn add @lendi-ui/typography
```

## Usage

- Body

```jsx
import { Body } from '@lendi-ui/typography';
<Body size="lg" color="warn.500">
  The quick brown fox jumps over the lazy dog
</Body>;
```

- Heading

```jsx
import { Heading } from '@lendi-ui/typography';
<Heading size="lg" color="warn.500">
  The quick brown fox jumps over the lazy dog
</Heading>;
```

- Link

```jsx
import { Link } from '@lendi-ui/typography';
<Link>quick brown fox</Link>;
```
