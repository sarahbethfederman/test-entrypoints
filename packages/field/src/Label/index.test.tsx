import * as React from 'react';
import { mount } from 'enzyme';
import Label from './index';
import { LabelBody, LabelHeading, AssistiveBody } from './index.style';
import { Link } from '@lendi-ui/typography';
import Theme from '@lendi-ui/theme';

let wrapper;

function render(props) {
  wrapper = mount(
    <Theme>
      <Label label="First name" {...props} />
    </Theme>
  );
}

describe('Large Label', () => {
  const size = 'sm';
  const link = <Link size="sm">forgot?</Link>;
  const assistiveText = 'Assistive text';
  it('it should render LUI "md" body component', () => {
    render({ size });
    expect(wrapper.find(LabelBody).length).toEqual(1);
    expect(wrapper.find('strong').length).toEqual(1);
    expect(wrapper.find(LabelBody).prop('size')).toEqual('md');
    expect(wrapper.find('strong').prop('children')[0]).toEqual('First name');
    expect(wrapper.find(LabelBody)).toMatchSnapshot();
  });

  it('it should render LUI "md" Link component', () => {
    render({ size, link });
    expect(wrapper.find(Link).length).toEqual(1);
    expect(wrapper.find(Link).prop('size')).toEqual('sm');
    expect(wrapper.find(Link).prop('children')).toEqual('forgot?');
    expect(wrapper.find(Link)).toMatchSnapshot();
  });

  it('it should render LUI "md" Body assistive text component', () => {
    render({ size, link, assistiveText });
    expect(wrapper.find(AssistiveBody).length).toEqual(1);
    expect(wrapper.find(AssistiveBody).prop('size')).toEqual('xs');
    expect(wrapper.find(AssistiveBody).prop('children')).toEqual('Assistive text');
    expect(wrapper.find(AssistiveBody)).toMatchSnapshot();
  });
});

describe('Large Label', () => {
  const size = 'lg';
  const link = <Link size="md">forgot?</Link>;
  const assistiveText = 'Assistive text';
  it('it should render LUI "md" heading component', () => {
    render({ size });
    expect(wrapper.find(LabelHeading).length).toEqual(1);
    expect(wrapper.find(LabelHeading).prop('size')).toEqual('md');
    expect(wrapper.find(LabelHeading).prop('children')[0]).toEqual('First name');
    expect(wrapper.find(LabelHeading)).toMatchSnapshot();
  });

  it('it should render LUI "md" Link component', () => {
    render({ size, link });
    expect(wrapper.find(Link).length).toEqual(1);
    expect(wrapper.find(Link).prop('size')).toEqual('md');
    expect(wrapper.find(Link).prop('children')).toEqual('forgot?');
    expect(wrapper.find(Link)).toMatchSnapshot();
  });

  it('it should render LUI "md" Body assistive text component', () => {
    render({ size, link, assistiveText });
    expect(wrapper.find(AssistiveBody).length).toEqual(1);
    expect(wrapper.find(AssistiveBody).prop('size')).toEqual('md');
    expect(wrapper.find(AssistiveBody).prop('children')).toEqual('Assistive text');
    expect(wrapper.find(AssistiveBody)).toMatchSnapshot();
  });
});
