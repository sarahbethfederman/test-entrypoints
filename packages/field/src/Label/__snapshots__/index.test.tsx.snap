// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Large Label should render LUI "md" Body assistive text component 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: rgba(146,146,146,1);
  font-size: calc(1.125 * var(--lendi-ui-size));
  line-height: calc(24 / 18);
  font-family: 'Open Sans',sans-serif;
  font-size: var(--lendi-ui-size);
}

<Styled(styled.p)
  color="shade.400"
  size="lg"
>
  <StyledComponent
    color="shade.400"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htoDjs",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
            "
  ",
            [Function],
            ";
",
          ],
        },
        "displayName": "Styled(styled.p)",
        "foldedComponentIds": Array [
          "sc-bwzfXH",
        ],
        "render": [Function],
        "styledComponentId": "sc-htoDjs",
        "target": "p",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    size="lg"
  >
    <p
      className="c0"
      color="shade.400"
      size="lg"
    >
      Assistive text
    </p>
  </StyledComponent>
</Styled(styled.p)>
`;

exports[`Large Label should render LUI "md" Link component 1`] = `
.c0 {
  font-size: 1em;
  border: none;
  padding: 0;
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
  font-weight: bold;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: currentColor;
  background-color: inherit;
}

.c0:hover,
.c0:focus,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<Component
  size="md"
>
  <styled.button
    size="md"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-size: 1em;
  border: none;
  padding: 0;
  ",
              [Function],
              " ",
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background-color: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "button",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="md"
    >
      <button
        className="c0"
        size="md"
      >
        forgot?
      </button>
    </StyledComponent>
  </styled.button>
</Component>
`;

exports[`Large Label should render LUI "md" heading component 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(2.3125 * var(--lendi-ui-size));
  font-weight: 400;
  line-height: calc(40 / 37);
  font-family: 'Cabin',sans-serif;
  font-size: calc(1.75 * var(--lendi-ui-size));
  margin-right: var(--lendi-ui-size);
  display: inline-block;
}

<Styled(Styled(HeadingTag))
  size="lg"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
            "
  ",
            [Function],
            " display: inline-block;
",
          ],
        },
        "displayName": "Styled(Styled(HeadingTag))",
        "foldedComponentIds": Array [
          "sc-htpNat",
        ],
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": [Function],
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="lg"
  >
    <HeadingTag
      className="c0"
      size="lg"
    >
      <h2
        className="c0"
        size="lg"
      >
        First name
         (Optional)
      </h2>
    </HeadingTag>
  </StyledComponent>
</Styled(Styled(HeadingTag))>
`;
